Modules/_asynciomodule.c

```diff
@@ -2679,9 +2679,10 @@ task_step_impl(TaskObj *task, PyObject *exc)
         return NULL;
     }
 
+    PyGenSendStatus gen_status = PYGEN_ERROR;
     if (exc == NULL) {
         if (PyGen_CheckExact(coro) || PyCoro_CheckExact(coro)) {
-            result = _PyGen_Send((PyGenObject*)coro, Py_None);
+            gen_status = _PyGen_Send_NameTBD((PyGenObject*)coro, Py_None, &result);
```

Comment:
Would not be better if `_PyGen_Send_NameTBD()` would call method `send()` internally for non-generator and non-coroutines?